tests/python/multidevice/test_dtensor.py

```diff
@@ -104,69 +104,67 @@ def define_fusion(fd: FusionDefinition):
-        d, e = config.d, config.e
-        inp = fd.define_tensor([-1, -1, e])
-        weight = fd.define_tensor([d * e, e], contiguity=True)
-        bias = fd.define_tensor([d * e], contiguity=True)
-        out = fd.ops.linear(inp, weight, bias)
-        fd.add_output(out)
+@dataclass
+class LinearConfig:
+    in_features: int
+    out_features: int
 
-    def define_linear_backward(config: LinearConfig, fd: FusionDefinition) -> None:
-        d, e = config.d, config.e
-        x = fd.define_tensor([-1, -1, e])
-        w = fd.define_tensor([d * e, e], contiguity=True)
-        grad = fd.define_tensor([-1, -1, d * e], contiguity=True)
-
-        grad_x = fd.ops.matmul(grad, w)
-
-        grad_flat_t = fd.ops.permute(fd.ops.reshape(grad, [-1, d * e]), [1, 0])
-        x_flat = fd.ops.reshape(x, [-1, e])
-        grad_w = fd.ops.matmul(grad_flat_t, x_flat)
-
-        grad_b = fd.ops.sum(grad, [0, 1])
-
-        fd.add_output(grad_x)
-        fd.add_output(grad_w)
-        fd.add_output(grad_b)
-
-    class LinearFunction(torch.autograd.Function):
-        @staticmethod
-        def forward(
-            ctx,
-            input: DTensor,
-            weight: DTensor,
-            bias: DTensor,
-        ):
-            d = weight.device_mesh.size()
-            e = input.size(-1)
-            op = FusionDefinitionWrapper(
-                partial(define_linear_forward, LinearConfig(d, e))
-            )
-            outputs = op([input, weight, bias])
-            ctx.save_for_backward(input, weight)
-            return outputs[0]
-
-        @staticmethod
-        def backward(ctx, grad_output: DTensor):
-            d = grad_output.device_mesh.size()
-            e = grad_output.to_local().size(-1)
-            op = FusionDefinitionWrapper(
-                partial(define_linear_backward, LinearConfig(d, e))
+
+def define_linear_forward(config: LinearConfig, fd: FusionDefinition) -> None:
+    e_in, e_out = config.in_features, config.out_features
+
+    inp = fd.define_tensor([-1, -1, e_in], contiguity=True)
+    weight = fd.define_tensor([e_out, e_in], contiguity=True)
+    out = fd.ops.linear(inp, weight)
+    fd.add_output(out)
+
+
+def define_linear_backward(config: LinearConfig, fd: FusionDefinition) -> None:
+    e_in, e_out = config.in_features, config.out_features
+
+    x = fd.define_tensor([-1, -1, e_in], contiguity=True)
+    w = fd.define_tensor([e_out, e_in], contiguity=True)
+    grad = fd.define_tensor([-1, -1, e_out], contiguity=True)
+
+    grad_x = fd.ops.matmul(grad, w)
+
+    grad_flat_t = fd.ops.permute(fd.ops.reshape(grad, [-1, e_out]), [1, 0])
+    x_flat = fd.ops.reshape(x, [-1, e_in])
+    grad_w = fd.ops.matmul(grad_flat_t, x_flat)
+
+    fd.add_output(grad_x)
+    fd.add_output(grad_w)
+
+
+class LinearFunction(torch.autograd.Function):
+    @staticmethod
+    def forward(
+        ctx,
+        input: DTensor,
+        weight: DTensor,
+    ):
+        op = FusionDefinitionWrapper(
```

Comment:
nit:
Can we use `e_out, e_in` for the sizes of the weight for easier readability?

Reply:
Do you mean 
```
e_out, e_in = weight.size()
partial(define_linear_forward, LinearConfig(e_out, e_in)))
```
?